### PR TITLE
Add maintainer documentation page

### PR DESCRIPTION
**Pull request by jrbourbeau**
This PR documents some of our maintainer practices that folks use but have yet to be written down. This will hopefully help when onboarding new maintainers. 

Currently this focuses on best practices around merging PRs. I'd like to start with this and then iterate on this page in follow-up PRs. 

cc @jsignell @jacobtomlinson @pavithraes
Also @ian-r-rose @fjetter @crusaderky (who are currently OOO)

### CODE REVIEW

File: docs/source/maintainers.rst
Comment (jsignell) [inline docs/source/maintainers.rst:20]:
```suggestion
merging. However, because reviewer bandwidth is limited, maintainers will
```

File: docs/source/maintainers.rst
Comment (jsignell) [inline docs/source/maintainers.rst:23]:
```suggestion
a substantial pull request which hasn't received a review, they should ping at
least one team or individual who might be interested in the proposed change. 
If after some time (maybe 2 days) there is no response and the maintainer
```

File: docs/source/maintainers.rst
Comment (jsignell) [inline docs/source/maintainers.rst:33]:
```suggestion
Use squash merging when merging pull requests, as opposed to other merging
```

File: docs/source/maintainers.rst
Comment (jsignell) [inline docs/source/maintainers.rst:62]:
```suggestion
2. Copy the original description of the pull request from the first comment or
   linked issue.
3. Leave the merge commit message blank (again, making sure the commit title is
```

**Review by jsignell — COMMENTED** [review]
This looks really good! I just tried to add a bit more about what we are currently doing. Love having this explicit!

File: docs/source/maintainers.rst
Comment (jsignell) [reply docs/source/maintainers.rst:28]: oh interesting! I read this as "don't merge if you don't understand the new code"

File: docs/source/maintainers.rst
Comment (ncclementi) [inline docs/source/maintainers.rst:21]:
```suggestion
sometimes self-merge their own pull requests if the content of the pull
```
content is viewed or contentS are viewed

File: docs/source/maintainers.rst
Comment (ncclementi) [inline docs/source/maintainers.rst:42]:
```suggestion
title and (optional) message meaningfully reflect the content of the pull
```

File: docs/source/maintainers.rst
Comment (jacobtomlinson) [reply docs/source/maintainers.rst:28]: @mrocklin made a point at SciPy that the person who hits merge should be available to handle any fallout in the short term. That's what I'm trying to capture here but may not be putting it into the right words.

Comment (mrocklin) [reply docs/source/maintainers.rst:28]: What I meant to say is that as we merge things we should have a plan for if they break and cause issues.  That plan could be colleagues or some other arrangement.  We probably shouldn't be merging things if we don't have such a plan though.

File: docs/source/maintainers.rst
Comment (jrbourbeau) [reply docs/source/maintainers.rst:23]: Ah, great. I made one minor tweak to the suggestion: adding the word "maintainer" to "or individual _maintainer_ who might be interested". Thoughts?

File: docs/source/maintainers.rst
Comment (jsignell) [reply docs/source/maintainers.rst:28]: Oh yeah I get what you all are saying. I was just trying to agree that the current language is unclear, because I understood it differently from you. I think your suggestion is good.

File: docs/source/maintainers.rst
Comment (jsignell) [reply docs/source/maintainers.rst:23]: In my mind the individual might not really be strictly a maintainer, I'm thinking an array expert for example, but I am happy with whatever.

**Comment by ncclementi** [comment]
just a heads up I'm merging main, to avoid CI stalling again.

**Comment by jsignell** [comment]
I think there is value it wrapping this up and getting it in! @jrbourbeau should we just commit my last two suggestions and merge?

**Comment by ian-r-rose** [comment]
I have no objections to this content (even if I have some personal disagreements about squash merges :wink: ), thanks for writing it up @jrbourbeau!

I wonder if some additional text about CI would be helpful. Something along the lines of:

> Before merging a pull request, maintainers should make every effort to ensure that CI passes.
> Often this will require looking into the logs of a failed run to see what went wrong, and alerting the pull-request author.
> Ideally, no pull-request should be merged if there are CI failures, as broken CI in ``main`` can easily
> mask *more* problems with other PRs, and over the long haul a consistently broken CI
> is demoralizing for maintainers. However, in the real world, there are sometimes flaky tests,
> broken upstream dependencies, and failures that are otherwise obviously not related to the PR at hand.
> If that is the case, a maintainer *may* merge a PR with failing tests, but they should be prepared
> to follow up with any failures that result from such an unsafe operation.

**Comment by jrbourbeau** [comment]
Thanks for the reviews all -- I'll go ahead and push up suggestions and then we should be good to merge this in and iterate

**Comment by pavithraes** [comment]
@jrbourbeau Thank you so much for writing this up! The CI failure is unrelated, and I think merging `main` will fix it. :)

File: docs/source/maintainers.rst
Comment (fjetter) [inline docs/source/maintainers.rst:25]: lazy consensus within 24hr is a bit tricky given different timezones and the fact that people are on PTO, are busy, etc. Typically one chooses rather 48h or even 72h to account for this.

File: docs/source/maintainers.rst
Comment (fjetter) [reply docs/source/maintainers.rst:23]: I think most of these situations occur because people simply do not know that there is a small PR that can be easily reviewed and merged.
How about instead tagging a team to get feedback faster?

File: docs/source/maintainers.rst
Comment (jrbourbeau) [reply docs/source/maintainers.rst]: Made this update -- thanks!

File: docs/source/maintainers.rst
Comment (jrbourbeau) [reply docs/source/maintainers.rst:82]: Thanks for the suggestion @jsignell. It looks like it is talking about a specific way in which one can generate their own merge commit message. Additionally, it could be that the original PR or issue description also aren't super informative. To keep instructions here simple, I'd prefer to just say (1)  write your own commit message (however you'd like) or (2) leave blank.

> I would actually suggest to always opt for leave empty in the spirit of standardization. The merge commit always links back to the PR so even taking over the PR description is somewhat redundant (assuming there is internet)

I personally opt to always leave blank, but in https://github.com/dask/community/issues/164 there were some who prefer to include a commit message (when they have time). I'll propose we keep what is currently in this PR as it reflects what we do today. Happy to have a conversation around changing things, but I'd prefer to do that outside the scope of this PR

File: docs/source/maintainers.rst
Comment (jrbourbeau) [reply docs/source/maintainers.rst]: Great point -- updated this to a 48 hour minimum

File: docs/source/maintainers.rst
Comment (jrbourbeau) [reply docs/source/maintainers.rst]: Based on @jsignell's feedback here https://github.com/dask/dask/pull/9309#discussion_r930087194 I updated this section to also include a "ping a team or individual maintainer" step.

I agree the ideal case is that all PRs are reviewed before being merged. This is also what happens for almost all PRs. That said, sometimes people are busy with other things and reviews are hard to come by. Similar to https://github.com/dask/dask/pull/9309#discussion_r951889892, I'd prefer to move forward with this section as is (since it accurately describes our current behavior) and, if desired, opening up a separate conversation around changing this practice

**Comment by jrbourbeau** [comment]
Alright, I believe I've incorporated or responded to everyone's suggestions, so this should be ready for a (possibly) final review. Florian brought up a couple of points that, while totally valid, I think merit a separate conversation outside of this PR.

**Review by pavithraes — APPROVED** [review]

**Review by jacobtomlinson — APPROVED** [review]

**Comment by jrbourbeau** [comment]
Thanks all for the reviews / suggestions. Going to merge this in and we can keep iterating

**Pull request merged**
